sample/samplers.go

```diff
@@ -123,6 +120,9 @@ func (s *Sampler) sample(tokens []token) (token, error) {
 		return 1
 	})
 
+	if math.IsNaN(float64(sum)) {
```

Comment:
If all logits are NaNs then the sum will be NaN - we should error in this case. Added a test case as well